src/databricks/labs/ucx/hive_metastore/tables.py

```diff
@@ -252,3 +267,105 @@ def _init_seen_tables(self):
+        logger.info(
+            f"Reverting {table.object_type} table {table.database}.{table.name} upgraded_to {table.upgraded_to}"
+        )
+        self._backend.execute(table.sql_unset_upgraded_to("hive_metastore"))
+        self._backend.execute(f"DROP {table.kind} IF EXISTS {target_table_key}")
+
+    def _get_revert_count(self, schema: str | None = None, table: str | None = None) -> list[MigrationCount]:
+        upgraded_tables = self._get_tables_to_revert(schema=schema, table=table)
+
+        table_by_database = defaultdict(list)
+        for cur_table in upgraded_tables:
+            table_by_database[cur_table.database].append(cur_table)
+
+        migration_list = []
+        for cur_database in table_by_database.keys():
+            external_tables = 0
+            managed_tables = 0
+            views = 0
+            for current_table in table_by_database[cur_database]:
+                if current_table.upgraded_to is not None:
+                    if current_table.kind == "VIEW":
+                        views += 1
+                        continue
+                    if current_table.object_type == "EXTERNAL":
+                        external_tables += 1
+                        continue
+                    if current_table.object_type == "MANAGED":
+                        managed_tables += 1
+                        continue
+            migration_list.append(
+                MigrationCount(
+                    database=cur_database, managed_tables=managed_tables, external_tables=external_tables, views=views
+                )
+            )
+        return migration_list
+
+    def is_upgraded(self, schema: str, table: str) -> bool:
+        result = self._backend.fetch(f"SHOW TBLPROPERTIES `{schema}`.`{table}`")
+        for value in result:
+            if value["key"] == "upgraded_to":
+                logger.info(f"{schema}.{table} is set as upgraded")
+                return True
+        logger.info(f"{schema}.{table} is set as not upgraded")
+        return False
+
+    def print_revert_report(self, *, delete_managed: bool) -> bool | None:
+        migrated_count = self._get_revert_count()
+        if not migrated_count:
+            logger.info("No migrated tables were found.")
+            return False
+        print("The following is the count of migrated tables and views found in scope:")
+        print("Database                      | External Tables  | Managed Table    | Views            |")
+        print("=" * 88)
+        for count in migrated_count:
+            print(f"{count.database:<30}| {count.external_tables:16} | {count.managed_tables:16} | {count.views:16} |")
+        print("=" * 88)
+        print("Migrated External Tables and Views (targets) will be deleted")
+        if delete_managed:
+            print("Migrated Manged Tables (targets) will be deleted")
+        else:
+            print("Migrated Manged Tables (targets) will be left intact")
```

Comment:
Explain how to call the app for managed tables to be deleted. Nobody reads docs - only error messages

Reply:
OK.